newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/weaver/ClassWeaverService.java

```diff
@@ -238,52 +237,43 @@ private Collection<ClassMatchVisitorFactory> loadInternalSecurityWeavePackages()
-                                    if (null == internalWeavePackage) {
-                                        LOG.log(Level.FINEST, "internal weave package: {0} was null", instrumentationUrl.toExternalForm());
-                                        continue;
-                                    } else if (internalWeavePackage.getPackageViolations().size() > 0) {
-                                        LOG.log(Level.FINER, "skip loading weave package: {0}", internalWeavePackage.getName());
-                                        for (WeaveViolation violation : internalWeavePackage.getPackageViolations()) {
-                                            LOG.log(Level.FINER, "\t violation: {0}", violation);
-                                        }
-                                    } else {
-                                        LOG.log(Level.FINER, "adding weave package: {0}", internalWeavePackage.getName());
-                                        LOG.log(Level.INFO, "adding weave package: {0}", internalWeavePackage.getName());
-
-                                        internalWeavePackages.add(internalWeavePackage.getName());
-                                        weavePackageManager.register(internalWeavePackage);
-                                    }
-                                } catch (Throwable t) {
-                                    LOG.log(Level.FINER, t, "unable to load weave package jar {0}", instrumentationUrl);
+            Runnable loadWeavePackagesRunnable = () -> {
+                try {
+                    for (final String name : weavePackageJars) {
+                        URL instrumentationUrl = new URL("jar:" + securityAgentUrl.toExternalForm() + "!/" + name);
+                        try (InputStream inputStream = instrumentationUrl.openStream()) {
+                            WeavePackage internalWeavePackage = createWeavePackage(inputStream, instrumentationUrl.toExternalForm());
+                            if (null == internalWeavePackage) {
+                                LOG.log(Level.FINEST, "internal weave package: {0} was null", instrumentationUrl.toExternalForm());
+                                continue;
+                            } else if (internalWeavePackage.getPackageViolations().size() > 0) {
+                                LOG.log(Level.FINER, "skip loading weave package: {0}", internalWeavePackage.getName());
+                                for (WeaveViolation violation : internalWeavePackage.getPackageViolations()) {
+                                    LOG.log(Level.FINER, "\t violation: {0}", violation);
                                 }
+                            } else {
+                                LOG.log(Level.FINER, "adding weave package: {0}", internalWeavePackage.getName());
+                                LOG.log(Level.INFO, "adding weave package: {0}", internalWeavePackage.getName());
```

Comment:
Not part of your changes, but I think we can do without one of this logs.

Reply:
Removed the `INFO` level logging. We only log at `FINER` in `loadInternalWeavePackages` and now `loadInternalSecurityWeavePackages` matches that.